common/presenters/events-to-timeline/event-details-helpers.test.js

```diff
@@ -0,0 +1,251 @@
+const proxyquire = require('proxyquire')
+
+const i18n = require('../../../config/i18n')
+
+const {
+  getMoveDetails,
+  getEventAgency,
+  getEventContext,
+  getEventRelationships,
+  getEventableDetails,
+  getEventProperties,
+} = proxyquire('./event-details-helpers', {
+  '../../../common/lib/api-client/models': {
+    event: {
+      fields: {
+        event_type: '',
+        prop_a: '',
+        prop_b: '',
+        details: '',
+        eventable: {
+          jsonApi: 'hasOne',
+        },
+        rel_a: {
+          jsonApi: 'hasOne',
+          type: 'rel_as',
+        },
+        rel_b: {
+          jsonApi: 'hasOne',
+          type: 'rel_bs',
+        },
+      },
+    },
+  },
+})
+
+describe('Timeline events', function () {
+  describe('Event details helpers', function () {
+    beforeEach(function () {
+      sinon.stub(i18n, 't').returnsArg(0)
+      sinon.stub(i18n, 'exists').returns(false)
+    })
+
+    context('#getMoveDetails', function () {
+      it('should return the move and person', function () {
+        const mockMove = { id: 'move', profile: { person: { id: 'person' } } }
+        const details = getMoveDetails(mockMove)
+        expect(details).to.deep.equal({
+          move: mockMove,
+          person: mockMove.profile.person,
+        })
+      })
+
+      it('should return just the move if it has no profile', function () {
+        const mockMove = { id: 'move' }
+        const details = getMoveDetails(mockMove)
+        expect(details).to.deep.equal({
+          move: mockMove,
+          person: undefined,
+        })
+      })
+    })
+
+    context('#getEventAgency', function () {
+      context('when the event has a supplier', function () {
+        it('should return supplier as the agency', function () {
+          const mockEvent = {
+            supplier: 'supplier',
+          }
+          const details = getEventAgency(mockEvent)
+          expect(details).to.deep.equal({
+            agency: 'supplier',
+          })
+        })
+      })
+
+      context(
+        'when the event has no supplier but has a default agency',
+        function () {
+          const mockEvent = { event_type: 'foo' }
+          let details
+          beforeEach(function () {
+            i18n.exists.returns(true)
+            details = getEventAgency(mockEvent)
+          })
+          it('should lookup the default agency', function () {
+            expect(i18n.exists).to.be.calledOnceWithExactly(
+              'events::foo.defaultAgency'
+            )
+          })
+
+          it('should get default agency', function () {
+            expect(i18n.t).to.be.calledOnceWithExactly(
+              'events::foo.defaultAgency'
+            )
+          })
+
+          it('should return pmu as the agency', function () {
```

Comment:
I think this is a typo and should probably be `should return the default agency`